src/instantx.cpp

```diff
@@ -251,7 +257,7 @@ void CInstantSend::Vote(CTxLockCandidate& txLockCandidate)
 //received a consensus vote
 bool CInstantSend::ProcessTxLockVote(CNode* pfrom, CTxLockVote& vote)
 {
-    LOCK2(cs_main, cs_instantsend);
+    // cs_main, cs_wallet and cs_instantsend should be already locked
```

Comment:
Besides having this in comments we should probably also check if locks are actually held i.e.:
```diff
+    AssertLockHeld(cs_main);
+#ifdef ENABLE_WALLET
+    if (pwalletMain)
+        AssertLockHeld(pwalletMain->cs_wallet);
+#endif
+    AssertLockHeld(cs_instantsend);
```